### PR TITLE
Move Map.TilesByDistance to MapGrid and un-static it.

### PR DESCRIPTION
**Pull request by pchote**
As a bonus, mods can now configure the 50 cell limit in their mod.yaml.

### CODE REVIEW

**Comment by obrakmann** [comment]
looks ok, :+1:

File: OpenRA.Game/Map/Map.cs [hunk 1]
```diff
@@ -1152,8 +1110,8 @@ public IEnumerable<CPos> FindTilesInAnnulus(CPos center, int minRange, int maxRa
 			if (maxRange < minRange)
 				throw new ArgumentOutOfRangeException("maxRange", "Maximum range is less than the minimum range.");
 
-			if (maxRange >= TilesByDistance.Length)
-				throw new ArgumentOutOfRangeException("maxRange", "The requested range ({0}) exceeds the maximum allowed ({1})".F(maxRange, MaxTilesInCircleRange));
+			if (maxRange >= Grid.TilesByDistance.Length)
+				throw new ArgumentOutOfRangeException("maxRange", "The requested range ({0}) exceeds the maximum allowed ({1})".F(maxRange, Grid.MaximumTileSearchRange));
```
Comment (obrakmann) [inline OpenRA.Game/Map/Map.cs]: Actually, now that this is configurable, you might want to mention the setting in the message, something like "The requested range ({0}) cannot exceed the value of MaximumTileSearchRange ({1})."

(Similar thing in `Teleport.cs` below.

**Comment by pchote** [comment]
Fixed.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/32c933322f0664401a7f21afb2d6e67c070bc9a0...5629b65077366b285721f2163bf23ddaaa6a5b20)

**Pull request merged**